### PR TITLE
[LPS-148474 & LPS-148475 & LPS-148476] [CP 2.0] Team members listing

### PR DESCRIPTION
**Pull request by iagochaves**
// TO DO

### CODE REVIEW

**Review by igordmf — APPROVED** [review]
Good job!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/SupportSeatColumnType.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	return (
+		<>
+			{hasAdministratorAccess && (
+				<ClayIcon
+					className="cp-team-members-support-seat-icon"
+					symbol="check-circle-full"
+				/>
+			)}
+		</>
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/SupportSeatColumnType.js:29]:
```suggestion
			{hasAdministratorAccess && (
				<ClayIcon
					className="cp-team-members-support-seat-icon"
					symbol="check-circle-full"
				/>
			)}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/SupportSeatColumnType.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayIcon from '@clayui/icon';
+import {ROLE_TYPES} from '../../../../../../../common/utils/constants';
+
+const SupportSeatColumnType = ({roles}) => {
+	const hasAdministratorAccess = roles.find(
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/SupportSeatColumnType.js:16]:
```suggestion
const SupportSeatColumnType = ({roles}) => {
	const hasAdministratorAccess = !!roles?.find(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/RoleColumnType.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+				(roleType) => roleType.key === role
+			);
+
+			if (roleToDisplay) {
+				rolesAccumulator.push(roleToDisplay.name);
+			}
+
+			return rolesAccumulator;
+		}, [])
+		.join(', ');
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/RoleColumnType.js:27]: The reducer could do this join internally?

```suggestion
	const filteredRoles = roles
		?.reduce((rolesAccumulator, role) => {
			const roleToDisplay = Object.values(ROLE_TYPES).find(
				(roleType) => roleType.key === role
			);

			if (roleToDisplay) {
				rolesAccumulator.push(roleToDisplay.name);
			}

			return rolesAccumulator;
		}, [])
		.join(', ');
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/NameColumnType.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {Liferay} from '../../../../../../../common/services/liferay';
+import {getMd5Hash} from '../../../../../utils/getMd5Hash';
+
+const GRAVATAR_URL = 'https://www.gravatar.com/avatar';
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/NameColumnType.js:16]: This URL should be good comes by properties.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/NameColumnType.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+				height={AVATAR_SIZE_IN_PX}
+				src={`${GRAVATAR_URL}/${getMd5Hash(emailAddress)}`}
+				width={AVATAR_SIZE_IN_PX}
+			/>
+		</div>
+	);
+};
+
+const NameColumnType = ({userAccount}) => {
+	const currentLoggedUserId = Number(Liferay.ThemeDisplay.getUserId());
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/NameColumnType.js:32]:
```suggestion
	const currentLoggedUserId = +Liferay.ThemeDisplay.getUserId();
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/NameColumnType.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+const GRAVATAR_URL = 'https://www.gravatar.com/avatar';
+const AVATAR_SIZE_IN_PX = 40;
+
+const Avatar = ({emailAddress}) => {
+	return (
+		<div className="cp-team-members-avatar mr-2">
+			<img
+				height={AVATAR_SIZE_IN_PX}
+				src={`${GRAVATAR_URL}/${getMd5Hash(emailAddress)}`}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/NameColumnType.js:24]: Should be good pass the return from this function to a variable before use it
```suggestion
				src={`${GRAVATAR_URL}/${emailAddressMD5}`}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/components/Header/index.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+			className={classNames(
+				'align-items-center bg-neutral-1 d-flex px-2 rounded mb-2',
+				{
+					'py-3': hasAdminAccess,
+					'py-4': !hasAdminAccess,
+				}
+			)}
+		>
+			<div className="align-items-center d-flex ml-auto">
+				{project.maxRequestors && (
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/components/Header/index.js:52]:
```suggestion
				{!!project?.maxRequestors && (
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+		const getAccountUserAccounts = async () => {
+			const {data} = await client.query({
+				query: getAccountUserAccountsByExternalReferenceCode,
+				variables: {
+					externalReferenceCode: project.accountKey,
+				},
+			});
+
+			if (data) {
+				const accountUserAccounts = data.accountUserAccountsByExternalReferenceCode.items.reduce(
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:43]:
```suggestion
				const accountUserAccounts = data.accountUserAccountsByExternalReferenceCode?.items?.reduce(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+					(userAccountsAccumulator, userAccount) => {
+						const currentAccountBrief = userAccount.accountBriefs.find(
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:45]:
```suggestion
						const currentAccountBrief = userAccount.accountBriefs?.find(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+							(accountBrief) =>
+								accountBrief.externalReferenceCode ===
+								project.accountKey
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:48]:
```suggestion
								project?.accountKey
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+						);
+						if (currentAccountBrief) {
+							userAccountsAccumulator.push({
+								...userAccount,
+								roles: currentAccountBrief.roleBriefs.map(
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:53]:
```suggestion
								roles: currentAccountBrief.roleBriefs?.map(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+			}
+
+			setIsLoadingUserAccounts(false);
+		};
+		getAccountUserAccounts();
+	}, [project.accountKey]);
+
+	const hasAdminAccess = useMemo(() => {
+		const currentUser = userAccounts?.find(
+			({id}) => id === Number(Liferay.ThemeDisplay.getUserId())
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:73]:
```suggestion
			({id}) => id === +Liferay.ThemeDisplay.getUserId()
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+				hasAdminAccess={hasAdminAccess}
+				project={project}
+				userAccounts={userAccounts}
+			/>
+
+			<Table
+				className="border-0 cp-team-members-table"
+				columns={getColumnsByUserAccess(hasAdminAccess)}
+				isLoading={isLoadingUserAccounts}
+				rows={userAccounts.map((userAccount) => ({
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:99]:
```suggestion
				rows={userAccounts?.map((userAccount) => ({
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+		<div className="pt-2">
+			<TeamMembersTableHeader
+				hasAdminAccess={hasAdminAccess}
+				project={project}
+				userAccounts={userAccounts}
+			/>
+
+			<Table
+				className="border-0 cp-team-members-table"
+				columns={getColumnsByUserAccess(hasAdminAccess)}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:97]: Should be good pass the return to a variable and use it here
```suggestion
				columns={columnsByUserAccess}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+				isLoading={isLoadingUserAccounts}
+				rows={userAccounts.map((userAccount) => ({
+					email: (
+						<p className="m-0 text-truncate">
+							{userAccount.emailAddress}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:102]:
```suggestion
							{userAccount?.emailAddress}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+						</p>
+					),
+					name: <NameColumnType userAccount={userAccount} />,
+					options: <OptionsColumnType />,
+					role: <RoleColumnType roles={userAccount.roles} />,
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:107]:
```suggestion
					role: <RoleColumnType roles={userAccount?.roles} />,
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+					status: (
+						<StatusColumnType
+							hasLoggedBefore={userAccount.lastLoginDate}
+						/>
+					),
+					supportSeat: (
+						<SupportSeatColumnType roles={userAccount.roles} />
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js:114]:
```suggestion
							hasLoggedBefore={userAccount?.lastLoginDate}
						/>
					),
					supportSeat: (
						<SupportSeatColumnType roles={userAccount?.roles} />
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/RoleColumnType.js (hunk 3 above)
Comment (iagochaves) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/utils/constants/columns-definitions/RoleColumnType.js]: This logic was removed, check the newest version

**Review by jairmedeiros — APPROVED** [review]

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by iagochaves** [comment]
re-sent https://github.com/pt-liferay-solutions/liferay-portal/pull/1001

**Pull request closed without merging**